pkg/entities/record.go

```diff
@@ -53,38 +53,38 @@ type baseRecord struct {
 }
 
 type dataRecord struct {
-	*baseRecord
+	baseRecord
 }
 
-func NewDataRecord(id uint16) *dataRecord {
+func NewDataRecord(id uint16, numElements int) *dataRecord {
 	return &dataRecord{
-		&baseRecord{
+		baseRecord{
 			buff:               bytes.Buffer{},
 			len:                0,
 			fieldCount:         0,
 			templateID:         id,
 			orderedElementList: make([]*InfoElementWithValue, 0),
-			elementsMap:        make(map[string]*InfoElementWithValue),
+			elementsMap:        make(map[string]*InfoElementWithValue, numElements),
 		},
 	}
 }
 
 type templateRecord struct {
-	*baseRecord
+	baseRecord
 	// Minimum data record length required to be sent for this template.
 	// Elements with variable length are considered to be one byte.
 	minDataRecLength uint16
 }
 
 func NewTemplateRecord(count uint16, id uint16) *templateRecord {
```

Comment:
We can change the order of parameters like `NewTemplateRecord(id uint16, numElements uint16)` similar to `NewDataRecord`